File: server/app/models/link.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class LinkCreate(BaseModel):
    url: str


class LinkResponse(BaseModel):
    id: str
    url: str
    title: str
    thumbnail: Optional[str] = None
    summary: str
    tags: list[str]
    category: Optional[str] = None
    user_id: str
    created_at: datetime


class LinkListResponse(BaseModel):
    links: list[LinkResponse]
    total: int

File: app/src/types/link.ts
export interface Link {
  id: string;
  url: string;
  title: string;
  thumbnail: string | null;
  summary: string;
  tags: string[];
  category: string | null;
  created_at: string;
}

export interface LinkListResponse {
  links: Link[];
  total: number;
}
